Algorithms/DataCompression/huffman.cs

```diff
@@ -27,184 +28,298 @@ public Huff(Huff leftChild, Huff rightChild)
 
     internal class Man
     {
-        public List<string> Codec { get; set; } = new List<string>();
-        public List<string> Data { get; set; } = new List<string>();
+        public List<string> Codec { get; } = new List<string>();
+        public List<string> Data { get; } = new List<string>();
     }
 
-    internal class Program
+    /// <summary>
+    /// Greedy lossless compression algorithm.
+    /// </summary>
+    public class HuffmanAlgorithm
     {
-        private static void Main()
+        private int Len { get; set; }
+        private int Pos { get; set; }
+
+        private int[] _d;
+        private int[] _d1;
+        private char[] _c;
+        private char[] _c1;
```

Comment:
Please make the names more descriptive. And it's better to use parameters instead of fields to store state if possible, this way the object will be safe to use in multi-threaded environment.